spec/gemfury/client_spec.rb

```diff
@@ -101,25 +101,56 @@
   end
 
   describe '#push_gems' do
-    let(:stub_api_method)  do
-      stub_put("uploads/WTFBBQ123", 2).to_return(:body => fixture('upload.json'))
-      stub_post("uploads", 2) # Do this last so we can extend it
+    let(:fixture_gem) do
+      fixture('fury-0.0.2.gem')
     end
 
-    it_should_behave_like 'API without authentication' do
-      let(:send_api_request) { @client.push_gem(['gemfile']) }
+    let(:stub_api_method) do
+      stub = stub_post(@client.account, :endpoint => Gemfury.pushpoint)
+
+      if @client.account
+        stub.with(:query => { :as => @client.account })
+      else
+        stub
+      end
+    end
+
+    let(:send_api_request) { @client.push_gem(fixture_gem) }
+
+    it_should_behave_like 'API without authentication'
+
+    shared_examples 'uploading gems' do
+      it 'should upload successfully' do
+        @client.push_gem(fixture_gem)
+
+        post_opts = { :endpoint => Gemfury.pushpoint }
+
+        if @client.account
+          post_opts[:query] = { :as => @client.account }
+        end
+
+        expect(a_post(@client.account, post_opts)).to have_been_made
```

Comment:
Path should be `"uploads"` here too

Reply:
Hi Michael, i am confused here.

I tried to use 'uploads' as a path in the Push API, and i think the back-end is treating the 'uploads' word as a username. What works is if there's no path, then i noticed, it will use the authenticating user's own account, and the API works fine.

Please let me know what is expected behavior here?

Thanks!

Mark